### PR TITLE
VZ-8349 use config map or secret for overrides in istio and nginx (#5…

### PR DESCRIPTION
**Pull request by AbeHern**
* Backport; Added logic to allow ConfigMap and Secret overrides and validating any externalIPs present

### CODE REVIEW

File: platform-operator/controllers/clusters/jaeger.go [hunk 1]
```diff
@@ -59,7 +58,7 @@ func (r *VerrazzanoManagedClusterReconciler) getJaegerOpenSearchConfig(vzList *v
 		return jc, nil
 	}
 	// If OpenSearch storage is not configured for the Jaeger instance, then just return
-	if jsc.storageType != "elasticsearch" {
+	if jsc.storageType != "opensearch" {
```
Comment (pshirodkar-orcl) [inline platform-operator/controllers/clusters/jaeger.go:61]: This change seems incorrect. I believe the storage type should remain "elasticsearch" for 1.4, right? Your original PR did not make such changes.

File: platform-operator/controllers/clusters/jaeger.go [hunk 2]
```diff
@@ -106,58 +105,58 @@ func (r *VerrazzanoManagedClusterReconciler) getJaegerSpecConfig(vzList *vzapi.V
 			jsc.jaegerCreate = true
 			jsc.OSURL = vzconstants.DefaultJaegerOSURL
 			jsc.secName = vzconstants.DefaultJaegerSecretName
-			jsc.storageType = "elasticsearch"
+			jsc.storageType = "opensearch"
```
Comment (pshirodkar-orcl) [inline platform-operator/controllers/clusters/jaeger.go:108]: same as above

**Comment by anders-swanson** [comment]
Can you please check the cherry pick contains the correct files?

File: platform-operator/controllers/verrazzano/component/helm/helm_component.go [hunk 3]
```diff
@@ -103,6 +104,8 @@ type HelmComponent struct {
 
 	// Certificates associated with the component
 	Certificates []types.NamespacedName
+
+	AvailabilityObjects *ready.AvailabilityObjects
```
Comment (AbeHern) [inline platform-operator/controllers/verrazzano/component/helm/helm_component.go:108]: Remove

File: platform-operator/controllers/verrazzano/component/helm/helm_component.go [hunk 4]
```diff
@@ -213,6 +216,16 @@ func (h HelmComponent) IsInstalled(context spi.ComponentContext) (bool, error) {
 	return installed, nil
 }
 
+// IsAvailable Indicates whether a component is available for end users
+// Components should implement comprehensive availability checks, supplying an appropriate reason
```
Comment (AbeHern) [inline platform-operator/controllers/verrazzano/component/helm/helm_component.go:220]: Remove

File: platform-operator/controllers/verrazzano/component/helm/helm_component.go [hunk 5]
```diff
@@ -458,7 +517,7 @@ func (h HelmComponent) buildCustomHelmOverrides(context spi.ComponentContext, na
 	// Sort the kvs list by priority (0th term has the highest priority)
 
 	// Getting user defined Helm overrides as the highest priority
-	overrideStrings, err := common.GetInstallOverridesYAML(context, h.GetOverrides(context.EffectiveCR()).([]v1alpha1.Overrides))
```
Comment (AbeHern) [inline platform-operator/controllers/verrazzano/component/helm/helm_component.go:461]: ONLY KEEP

File: cluster-operator/THIRD_PARTY_LICENSES.txt [hunk 6]
```diff
@@ -0,0 +1,4131 @@
+
```
Comment (bobdonat) [inline cluster-operator/THIRD_PARTY_LICENSES.txt:1]: Did you run the attribution-helper which regenerated this?  I would not expect this file to change.

File: cluster-operator/THIRD_PARTY_LICENSES.txt
Comment (anders-swanson) [inline cluster-operator/THIRD_PARTY_LICENSES.txt:1]: why is this file included?

Comment (AbeHern) [reply cluster-operator/THIRD_PARTY_LICENSES.txt:1]: Oh it wasn't suppose to be, let me remove it

File: cluster-operator/THIRD_PARTY_LICENSES.txt (hunk 6 above)
Comment (AbeHern) [reply cluster-operator/THIRD_PARTY_LICENSES.txt:1]: Need to remove it, wasn't suppose to be added

**Review by bobdonat — APPROVED** [review]

**Review by anders-swanson — APPROVED** [review]
